Makefile

```diff
@@ -0,0 +1,232 @@
+BASE_PACKAGES := utils db settings visualization
+
+MODELS := biocro clm45 dalec ed gday jules linkages \
+	      lpjguess maat maespa preles sipnet
+
+
+MODULES := allometry assim.batch assim.sequential benchmark \
+	       data.atmosphere data.hydrology data.land \
+	       data.mining data.remote emulator meta.analysis \
+	       photosynthesis priors rtm uncertainty
+
+.PHONY: all ${BASE_PACKAGES} ${MODELS} ${MODULES}
+
+all: .install PEcAn.all ${BASE_PACKAGES} ${MODELS} ${MODULES}
+
+clean:
+	rm -rf .install
+
+.install:
+	mkdir -p $@
+
+#### DEPENDENCY MAPPING ####
+
+PEcAn.all: ${BASE_PACKAGES} ${MODULES} ${MODELS}
+
+${MODELS}: ${BASE_PACKAGES} ${MODULES}
+
+${BASE_PACKAGES} ${MODELS} ${MODULES}: .install/devtools
+
+.install/devtools:
+	Rscript -e "req <- require('devtools'); if(!req) install.packages('devtools')"
```

Comment:
this failed in a headless environment. I needed to add repos='http://cran.rstudio.com/'